vscode/src/repository/repo-name-resolver.test.ts

```diff
@@ -0,0 +1,146 @@
+import { describe, expect, it, vi } from 'vitest'
+import * as vscode from 'vscode'
+import { URI } from 'vscode-uri'
+
+import dedent from 'dedent'
+import { gitRemoteUrlFromTreeWalk } from './repo-name-resolver'
+
+describe('gitRemoteUrlFromTreeWalk', () => {
+    it('finds the remote url in the .git/config file with one remote', async () => {
+        const mockGitConfig = dedent`[core]
+            repositoryformatversion = 0
+            filemode = true
+            bare = false
+            logallrefupdates = true
+            ignorecase = true
+        [remote "origin"]
+            url = https://github.com/sourcegraph/cody
+            fetch = +refs/heads/*:refs/remotes/origin/*
+        [branch "main"]
+            remote = origin
+            merge = refs/heads/main
+        `
```

Comment:
Not important but dedent will trim the text (if I’m not completely wrong) so I think this is a little nicer to read :)
```suggestion
        const mockGitConfig = dedent`
        [core]
            repositoryformatversion = 0
            filemode = true
            bare = false
            logallrefupdates = true
            ignorecase = true
        [remote "origin"]
            url = https://github.com/sourcegraph/cody
            fetch = +refs/heads/*:refs/remotes/origin/*
        [branch "main"]
            remote = origin
            merge = refs/heads/main
        `
```